test/unit/math/fwd/mat/fun/append_array_test.cpp

```diff
@@ -3,10 +3,11 @@
 #include <gtest/gtest.h>
 
 using namespace Eigen;
```

Comment:
This isn't about this line number, but the tests in general.  Overally, they look OK, but you want to pull repeated functionality out into functions as much as possible.  We haven't done a good job of that in tests in the past.

I don't see any `*/arr/` tests for `std::vector` or the scalar tests.  You need to make sure that int and double casting works right as well as through `std::vector<int>` and `std::vector<double>`.